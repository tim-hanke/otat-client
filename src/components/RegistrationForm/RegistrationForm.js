import React, { Component } from "react";
import AuthApiService from "../../services/auth-api-service";
import TokenService from "../../services/token-service";
import { Button, Input, Required } from "../Utils/Utils";

export default class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {},
  };

  state = { error: null };

  handleSubmit = async (ev) => {
    ev.preventDefault();
    const { full_name, user_name, password } = ev.target;

    this.setState({ error: null });

    try {
      const user = await AuthApiService.postUser({
        full_name: full_name.value,
        user_name: user_name.value,
        password: password.value,
      });
      const res = await AuthApiService.postLogin({
        ...user,
        password: password.value,
      });
      TokenService.saveAuthToken(res.authToken);
      full_name.value = "";
      user_name.value = "";
      password.value = "";
      this.props.onRegistrationSuccess();
    } catch (err) {
      this.setState({ error: err.error });
    }
  };

  render() {
    const { error } = this.state;
    return (
      <form className="RegistrationForm" onSubmit={this.handleSubmit}>
        <div className="full_name">
          <label htmlFor="RegistrationForm__full_name">
            Full name <Required />
          </label>
          <Input
            name="full_name"
            type="text"
            required
            id="RegistrationForm__full_name"
          ></Input>
        </div>
        <div className="user_name">
          <label htmlFor="RegistrationForm__user_name">
            User name/email <Required />
          </label>
          <Input
            name="user_name"
            type="text"
            required
            id="RegistrationForm__user_name"
          ></Input>
        </div>
        <div className="password">
          <label htmlFor="RegistrationForm__password">
            Password <Required />
          </label>
          <Input
            name="password"
            type="password"
            required
            id="RegistrationForm__password"
          ></Input>
        </div>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <Button type="submit">Register</Button>
      </form>
    );
  }
}
