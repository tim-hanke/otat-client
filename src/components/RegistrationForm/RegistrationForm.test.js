import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import RegistrationForm from "./RegistrationForm";

describe("RegistrationForm component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<RegistrationForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<RegistrationForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
