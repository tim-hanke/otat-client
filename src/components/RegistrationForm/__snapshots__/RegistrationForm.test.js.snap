// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders the UI as expected 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    className="full_name"
  >
    <label
      htmlFor="RegistrationForm__full_name"
    >
      Full name 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__full_name"
      name="full_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      User name/email 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__user_name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div
    role="alert"
  />
  <button
    className="Button "
    type="submit"
  >
    Register
  </button>
</form>
`;
